.github/workflows/canary-integration-test.yml

```diff
@@ -108,6 +108,10 @@ jobs:
         name: canary
         path: test
 
+    - name: setup tmate session for debugging
+      if: failure() && contains(github.event.pull_request.labels.*.name, 'debug-canary')
```

Comment:
changes look good.
One questionm, how we are using this
`contains(github.event.pull_request.labels.*.name, 'debug-canary')`?

Reply:
I don't understand the question, can you clarify?

Reply:
so we can directly check `if: failure()` why this extra check `&& contains(github.event.pull_request.labels.*.name, 'debug-canary')`?

Reply:
The extra check is there so that we trigger the tmate on purpose with a label. We don't want to let the tmate run every time there is a failure. If we were doing this all the time we will have runners idle for a long time and then reduce our ability to run new runners.

Reply:
Understood. Thanks